### PR TITLE
chore(DLI): add checkdeleted and  *Context

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dli' TESTARGS='-run=TestAccDliQueue'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run=TestAccDliQueue -timeout 360m -parallel 4
=== RUN   TestAccDliQueue_basic
=== PAUSE TestAccDliQueue_basic
=== RUN   TestAccDliQueue_cidr
=== PAUSE TestAccDliQueue_cidr
=== CONT  TestAccDliQueue_basic
=== CONT  TestAccDliQueue_cidr
--- PASS: TestAccDliQueue_basic (254.57s)
--- PASS: TestAccDliQueue_cidr (263.47s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       263.517s
```

### CODE REVIEW

File: huaweicloud/services/dli/resource_huaweicloud_dli_queue.go [hunk 1]
```diff
@@ -216,70 +220,68 @@ func resourceDliQueueRead(d *schema.ResourceData, meta interface{}) error {
...
 
 		for _, v := range queueList.Queues {
 			if v.QueueName == queueName {
 				return &v, nil
 			}
 		}
-		return nil, nil
+		return nil, golangsdk.ErrDefault404{}
 
 	} else {
```
Comment (ShiChangkuo) [inline huaweicloud/services/dli/resource_huaweicloud_dli_queue.go]: else is unnecessary

Comment (hmbig2) [reply huaweicloud/services/dli/resource_huaweicloud_dli_queue.go]: done

**Pull request merged**